src/app/activity/bounty/Claim.ts

```diff
@@ -1,21 +1,43 @@
 import { GuildMember, Message, MessageEmbed, TextChannel } from 'discord.js';
 import { ClaimRequest } from '../../requests/ClaimRequest';
 import { BountyCollection } from '../../types/bounty/BountyCollection';
+import { Bounty } from '../../types/bounty/Bounty';
 import DiscordUtils from '../../utils/DiscordUtils';
 import Log, { LogUtils } from '../../utils/Log';
-import mongo, { Db, UpdateWriteOpResult } from 'mongodb';
+import mongo, { Cursor, Db, UpdateWriteOpResult } from 'mongodb';
 import MongoDbUtils from '../../utils/MongoDbUtils';
 import { CustomerCollection } from '../../types/bounty/CustomerCollection';
 import RuntimeError from '../../errors/RuntimeError';
 import { BountyEmbedFields } from '../../constants/embeds';
 import { BountyStatus } from '../../constants/bountyStatus';
+import BountyUtils from '../../utils/BountyUtils';
 
 export const claimBounty = async (request: ClaimRequest): Promise<any> => {
     const claimedByUser = await DiscordUtils.getGuildMemberFromUserId(request.userId, request.guildId);
-	Log.info(`${request.bountyId} bounty claimed by ${claimedByUser.user.tag}`);
-	
-    const getDbResult: {dbBountyResult: BountyCollection, bountyChannel: string} = await getDbHandler(request);
-    await writeDbHandler(request, claimedByUser);
+    Log.info(`${request.bountyId} bounty claimed by ${claimedByUser.user.tag}`);
+    
+    let getDbResult: {dbBountyResult: BountyCollection, bountyChannel: string} = await getDbHandler(request);
+
+    // If this is evergreen, see if this user already claimed an instance
```

Comment:
could you move this authorization logic to the authorization handler `AuthorizationModule::claim`

Reply:
Done